models/classes/tasks/ResultTransmission/Exception/ResultTransmissionTaskValidationException.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\classes\tasks\ResultTransmission\Exception;
+
+class ResultTransmissionTaskValidationException extends \Exception
```

Comment:
I'd say this kind of an exception should either be extending a `DomainException` or a `RuntimeException`.
I think it should extend the first one, as such an exception means 2 things:
1. The value was correctly typed, but the contents were corrupted
2. Some piece of code was likely to be responsible for the corrupted data, and has to be fixed, which is a definition of a `LogicException`

Reply:
good point